### DISCUSSION TITLE
I have an ST7789 with touch problems that doesn't work, I've read all the warnings, but despite all the tests I haven't solved it

### DISCUSSION THREAD

**Discussion by suntchi2018** [Bodmer/TFT_eSPI#3142] [Q&A - General]
Hi to all, sorry me, I was so focused on finding the solution to the problem that I forgot to say that I am using version 2.5.34 and that I thank Bodmer for his fantastic library that I have been using for some time with great satisfaction and above all I thank him for the excellent support that I have already successfully received other times previously.
This is the problem, I have 2 ST7789 displays, one just unpacked and tested, but in both the touch doesn't work. This is my setup. User_setup.h:
```
#define ST7789_DRIVER
#define TFT_RGB_ORDER TFT_BGR
#defines TFT_WIDTH 240
#defines TFT_HEIGHT 320
#define TFT_MISO 19
#define TFT_MOSI 23
#define TFT_SCLK 18
#define TFT_CS 15
#define TFT_DC 2
#define TFT_RST 4
#define TOUCH_CS 21

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_FONT8N
#define LOAD_GFXFF
#define SMOOTH_FONT
#define SPI_READ_FREQUENCY 5000000
#define SPI_TOUCH_FREQUENCY 2500000
```

The connections are correct and rechecked several times, on both displays and cables they are all working.
I tried the various tests; Colour_Test, Read_User_Setup, FTF_ReadWrite_Test which all work except the Test_Touch_Controller. The touch chip (controlled) is XPT2046.
Both displays work perfectly and the colors are correct.
The module in use is an ESP32-WROOM-32 but slightly different as it uses a different serial chip using USB-C (I attach a photo) the pining is identical the module chosen with Arduino IDE 2.2.1 is ESP32-WROOM-DA-Module.
If I touch the pins of the chip with my finger during the Test_TouchWrite_Test, I see some numbers so I guess the chip is working, maybe the connections are broken, but I doubt in both displays. Unfortunately the only possible configuration is the one indicated for the display with the ILI9341 chip but I imagine they are also valid for the ST7789 display.
I tried removing the MISO pin from the TFT connector, but it still doesn't work.

I attach photos of the chip.
![ESP32-WROOM-32-USB-C](https://github.com/Bodmer/TFT_eSPI/assets/40439798/00282b67-e2b9-46b4-bbfe-04ce5b03d66f)
![IMG_20240128_092732](https://github.com/Bodmer/TFT_eSPI/assets/40439798/6d634b44-923a-4877-ada1-373191a3dbe0)
![IMG_20240128_090149](https://github.com/Bodmer/TFT_eSPI/assets/40439798/54dac6e6-9a9d-4155-ad55-7d19dbb26bcf)
![esp32 wroom-tft-espi](https://github.com/Bodmer/TFT_eSPI/assets/40439798/9ff53c8d-0d56-444b-a0d4-18896e22ac1d)

**Comment by Bodmer** [comment]
SDI (Serial Data In = MOSI) of the display should connect to the touch DIN (Data In) pin and the display SDO (Serial Data Out = MISO) pin should connect to the touch DO (Data Out) pin.

**Reply by suntchi2018** [reply]
I apologize for the oversight, I checked it at least 100 times without ever realizing the error. Getting old is not nice, you realize it when these unpleasant situations happen to you.
I humbly apologize and thank you from the bottom of my heart for your amazing support and enormous patience, you are a magician.
I'm reposting the connection diagram that I corrected and I apologize again.
Now works
![esp32 wroom-tft-espi](https://github.com/Bodmer/TFT_eSPI/assets/40439798/29177379-7782-42e8-be5c-d8263c1e1718)

**Reply by Bodmer** [reply]
Glad to help and to hear it is working now. I hope you have fun with the library.

**Comment by suntchi2018** [comment]
Of course Bodmer, I have been using your library for a long time, I love it, it's a shame that although I don't lack ideas, I lack attention. I hope I won't bother you anymore with my oversights and I thank you and wish you much happiness.
